docs/cross-chain.md

```diff
@@ -2,46 +2,171 @@
+When a user wants to interact with a message queue, they create a request cell instead of directly writing into the queue cell. This is achieved by replacing the lock field of their own cell:
 
-1. Call `CSVBurn` on the source chain to burn `X` amount of tokens and provide a `mint out_point` on the destination chain.
-2. Wait for confirmation period.
-3. Generate a CSV proof on the source chain.
-4. Call `CSVMint` on the destination chain and provide the `CSV proof`.
-5. The `mint out_point` must be used as one of the mint transaction's inputs, and X amount of tokens will be minted.
+Batch Request Lock Fields
+
+- **cancel_timeout**: Specifies the number of blocks after which the request can be canceled.
+- **owner_lock_hash**: Allows the request to be canceled with this lock hash.
+- **request**: Serialized byte structure containing the request details.
+
+By creating request cell, users permit aggregators to collect and batch these requests. This approach allows multiple users to interact with the message queue simultaneously and efficiently.
+
+### RGB++ message queue
+
+The RGB++ side of the message queue is structured as follows:
+
+``` rust
+pub struct CKBMessageQueue {
+  // The merkle tree root of locked assets
+  locked_assets: Byte32,
+  // Send queue
+  outbox: Vec<CKBMessage>,
+  // Receive queue
+  inbox: Vec<BranchMessage>,
+}
+```
+
+Key operations include:
+
+* PushRequest: Submit cross chain transfer message.
+* Ack: Confirms messages have been processed by providing branch chain block headers and proofs.
+* HandleRecv: Handles the receive queue.
+
+### Branch message queue
+
+The branch chain side of the message queue is structured as follows:
+
+``` rust
+pub struct BranchMessageQueue {
+  // The merkle tree root of locked assets
+  locked_assets: Byte32,
+  // Send queue
+  outbox: Vec<Bytes(BranchMessage)>,
+  // Receive queue
+  inbox: Vec<CKBMessage>,
+}
+```
+
+Key operations include:
+
+* PushRequest: Submit cross chain transfer message.
+* HandleRecv: Handles the receive queue.
+
+
+### Messages
+
+``` rust
+/// CKB messages
+pub enum CKBMessage {
+  Branch(CKBToBranchRequest)
+}
+
+pub struct CKBToBranchRequest {
+  request_id: Bytes,
+  // Target chain ID
+  target_chain_id: ChainId,
+  message: Message,
+}
+
+// Branch chain messages
+pub enum BranchMessage {
+  CKB(BranchToCKBRequest)
+  Branch(BranchToBranchRequest)
+}
+
+pub struct BranchToCKBRequest {
+  request_id: Bytes,
+  // Initial chain ID
+  initial_chain_id: ChainId,
+  message: Message,
+}
+
+pub struct BranchToBranchRequest {
+  request_id: Bytes,
+  // Initial chain ID
+  initial_chain_id: ChainId,
+  // Target chain ID
+  target_chain_id: ChainId,
+  message: Message,
+}
+
+
+pub struct Message {
+  owner_lock_hash: Byte32,
+  amount: u128,
+  // Asset type
+  asset_type: Byte32,
+}
+```
+
+## Leap
+
+We use UTXO token as an example to describe the life cycle of cross chain assets.
+
+### RGB++ to Branch chain
```

Comment:
```suggestion
### RGB++ layer to Branch chain
```
IMO, RGB++ is a protocol by default.
So I think it's better to name CKB as RGB++ layer.